pagegraph/src/graph.rs

```diff
@@ -268,6 +282,8 @@ impl PartialEq for Edge {
 }
 
 #[derive(PartialEq, Eq, Hash, Clone, Copy, PartialOrd, Ord)]
+#[derive(serde::Serialize)]
+
```

Comment:
nit: empty line here and above (line 149), plus you could also combine these into the existing `#[derive(...)]` annotations: